### PR TITLE
Changelog sections customization

### LINKED ISSUE #193: Changelog is incomplete
Hi, thanks for this project, in first place.

I want to start using this project but I'm having difficulties in generating a proper changelog.

We _just started_ using commitizen. So there's thousands of commits before the ones using commitizen. We have about 12 commits using commitizen. We have `build`, `fix`, `revert`, `feat`, `fix`, `refactor` and `improvement.

Perhaps I'm misunderstanding how to use it, but the changelog has old commits. I'm guessing it's because they just so happen to be matched by the parsing. This is fine, I reckon.

But the changelog does not contain all of the recent commits already made using commitizen, only `fix` appear in the changelog. 

Using the DEBUG env var shows things like:

```
semantic_release:history:logs 'ignored commit' UnknownCommitMessageStyleError('Unable to parse the given commit message: revert(automation): remove outdated playbook\n') +1.1ms
```

```
semantic_release:history:logs 'ignored commit' UnknownCommitMessageStyleError('Unable to parse the given commit message: improvement(apiserver): pATCH on accounts now allow for updating protecte\n\nWhen instagram accounts are authenticated as Business accounts, their IDs change. This change allows\nsm_account_id changes for instagram accounts.\n\nre #5230\n') +1.9ms
```

Others are successful on the parsing:

```
semantic_release:history:parser_angular "parse_commit_message -> (0, refactor, apiserver,connector,sorter, ('simplify logic on detecting business instagram accounts', 'As Basic Display API is not supported, the need for detecting what kind of API the account should use has been simplified. ', ''))" +259.0us
```

_But they still don't show in the changelog_.

For test, I used it for the first time - bumping version, etc. Then made a few test commits and did the same. Not all commits show.

My setup.cfg is barebones:

```
[semantic_release]
version_variable = src/sm2/__init__.py:__version__
branch = develop  # testing
```

### PR DESCRIPTION
**Pull request by nicholasamorim**
Customization of changelog can be done via changelog_sections var in configuration.

Fixes #193

### CODE REVIEW

File: semantic_release/history/logs.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 # Sections which will be shown in the Markdown changelog.
 # This is NOT related to supported commit types.
-CHANGELOG_SECTIONS = [
+DEFAULT_CHANGELOG_SECTIONS = [
```
Comment (AlphaMycelium) [inline semantic_release/history/logs.py:22]: Why not use `defaults.cfg`?

File: semantic_release/history/logs.py [hunk 2]
```diff
@@ -173,7 +173,14 @@ def markdown_changelog(version: str, changelog: dict, header: bool = False) -> s
...
-    for section in CHANGELOG_SECTIONS:
+    custom_changelog_sections = config.get("semantic_release", "changelog_sections", fallback="")
+    if custom_changelog_sections:
+        debug('markdown_changelog(custom_changelog={})'.format(custom_changelog_sections))
+        changelog_sections = [s.strip() for s in custom_changelog_sections.split(',')]
+    else:
+        changelog_sections = DEFAULT_CHANGELOG_SECTIONS
+
+    for section in changelog_sections:
         if not changelog[section]:
```
Comment (AlphaMycelium) [inline semantic_release/history/logs.py:184]: Maybe add a check to skip invalid sections:

```suggestion
        if section not in changelog or not changelog[section]:
```

File: semantic_release/history/logs.py (hunk 1 above)
Comment (nicholasamorim) [reply semantic_release/history/logs.py:22]: Ah, I'm just not familiar with the code, will change.

File: semantic_release/history/logs.py (hunk 2 above)
Comment (nicholasamorim) [reply semantic_release/history/logs.py:184]: Will do

**Comment by AlphaMycelium** [comment]
Ah, the release workflow failed because I merged the PR, so it doesn't have permission to update `master` because it is protected. I'll leave merging to @relekang next time.

**Comment by relekang** [comment]
Hmm, I guess the same would happen for me. There is no difference on our rights to push to master. We might need to add github actions as an excemption 🤔

**Comment by relekang** [comment]
I removed the protection on master for now.

**Pull request merged**